src/Directory.Build.props

```diff
@@ -39,6 +39,14 @@
     <UpdateXlfOnBuild>true</UpdateXlfOnBuild>
```

Comment:
@dsplaisted Why is this hard-coded to true? It should be false in CI.

Reply:
cc @cdmihai I see it's from https://github.com/tmat/msbuild/commit/677197e28c8f999d5d4f10a72dbd59fd0d49c8e1

You're never going to get an out-of-date error like this and the build machine will be editing source files, which is bad.

If you want to do the updating manually before each hand-off, you can do ErrorOnOutOfDateXlf=false. And UpdateXlfOnBuild=false always.

Reply:
Is there any issue with these settings?

With these settings, people would include the xlf diffs with the PR that also changes the resx, so there should be no source changes on the build machine. Besides, our CI job is set to [(hopefully) fail](https://github.com/Microsoft/msbuild/blob/master/build/build.ps1#L276-L285) if there is any output out of git status.
I like `UpdateXlfOnBuild=true` because stuff just happens. Otherwise, the build would fail saying the xlf is out of sync, and then people would have to understand what that error means and how to sync the xlf. `ErrorOnOutOfDateXlf=true` is just a sanity check :)

Reply:
The recommendation for this workflow is to set UpdateXlfOnBuild to true but false on CI. This is how all of our other repos work. If you have a separate git status check, it's equivalent. But I would still prefer that you not set it unconditionally so that nobody without the git status check copies from you incorrectly.